### PR TITLE
Fix tests, add '.' and '!' to kwdopchar

### PR DESCRIPTION
**Pull request by jaredly**
Back in the "async await" discussions in reason land, there were vigorous proponents of both the "let." and "let!", so I figure I might as well add them here :)

### CODE REVIEW

File: formatTest/typeCheckedTests/expected_output/features408re.4.08.0.re [hunk 1]
```diff
@@ -0,0 +1,16 @@
+let (let.opt) = (x, f) =>
```
Comment (anmonteiro) [inline formatTest/typeCheckedTests/expected_output/features408re.4.08.0.re:1]: wait, why can't we put this on the other file too?

Comment (jaredly) [reply formatTest/typeCheckedTests/expected_output/features408re.4.08.0.re:1]: because it's not valid ocaml syntax :)

**Comment by jaredly** [comment]
ugh the tests are failing because a newline is different 🙃

**Comment by anmonteiro** [comment]
I tried to fix it on the Reason PR

**Pull request merged**